bin/varnishd/cache/cache_fetch.c

```diff
@@ -269,7 +269,10 @@ vbf_stp_mkbereq(struct worker *wrk, struct busyobj *bo)
 	}
 	http_ForceField(bo->bereq0, HTTP_HDR_PROTO, "HTTP/1.1");
 
-	if (bo->stale_oc != NULL &&
+	if (bo->stale_oc != NULL)
+		HSH_CheckBans(wrk, bo->stale_oc);
```

Comment:
Can you please explain why you think the additional BAN check(s) are necessary? My understanding is what we check bans in `HSH_Lookup()` and only get to the backend side with a `stale_oc` that survived the ban checks.

Reply:
I rebased against current master and updated the commit log of the entire patch series.